jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/AzureFetchOrchestrator.java

```diff
@@ -0,0 +1,108 @@
+package com.tmobile.pacbot.azure.inventory;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.management.Azure.Authenticated;
+import com.microsoft.azure.management.resources.Subscription;
+import com.tmobile.pacbot.azure.inventory.auth.AzureCredentialProvider;
+import com.tmobile.pacbot.azure.inventory.file.AssetFileGenerator;
+import com.tmobile.pacbot.azure.inventory.file.S3Uploader;
+import com.tmobile.pacbot.azure.inventory.vo.SubscriptionVH;
+
+@Component
+public class AzureFetchOrchestrator {
+	
+	@Autowired
+	AssetFileGenerator fileGenerator;
+	
+	@Autowired
+	AzureCredentialProvider azureCredentialProvider;
+	
+	/** The s 3 uploader. */
+	@Autowired
+	S3Uploader s3Uploader;
+	
+	@Value("${file.path}")
+	private String filePath ;
+
+	@Value("${tenants:}")
+	private String tenants;
+	
+	@Value("${s3}")
+	private String s3Bucket ;
+	
+	@Value("${s3.data}")
+	private String s3Data ;
+	
+	@Value("${s3.processed}")
+	private String s3Processed ;
+	
+	@Value("${s3.region}")
+	private String s3Region ;
+	
+	/** The log. */
+	private static Logger log = LoggerFactory.getLogger(AzureFetchOrchestrator.class);
+	
+	public Map<String, Object> orchestrate(){
+		
+		try{
+			List<SubscriptionVH> subscriptions = fetchSubscriptions();
+			if(subscriptions.isEmpty()){
+				ErrorManageUtil.uploadError("all", "all", "all", "Error fetching subscription Info ");
+				return ErrorManageUtil.formErrorCode();
+			}
+			
+			log.info("Start : FIle Generation");
+			fileGenerator.generateFiles(subscriptions,filePath);
+			log.info("End : FIle Generation");
+			
+			log.info("Start : Backup Current Files");
+			s3Uploader.backUpFiles(s3Bucket, s3Region, s3Data, s3Processed+ "/"+ new SimpleDateFormat("yyyyMMdd-HHmmss").format(new Date()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you don’t call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.